app/lib/S3Actions.scala

```diff
@@ -9,16 +9,25 @@ import com.amazonaws.services.s3.AmazonS3Client
 import com.amazonaws.services.s3.model.{ObjectMetadata, PutObjectRequest, PutObjectResult, AmazonS3Exception}
 import com.gu.pandomainauth.model.User
 
-case class S3Upload(url: URI)
+trait S3UploadResponse {
+  def fileName : String
+  def msg  : Option[String]
+}
+
+case class S3UploadSuccess(url: URI, fileName: String = null, msg: Option[String] = null)
+  extends S3UploadResponse
 
-object S3Upload {
+case class S3UploadFailure(fileName: String = null, msg: Option[String] = null)
+  extends S3UploadResponse
+
+object S3UploadResponse {
   def build(putObjectRequest: PutObjectRequest) = {
     val uri = Config.stage match {
       case "PROD" => s"https://uploads.guim.co.uk/${putObjectRequest.getKey}"
       case _ => s"https://${putObjectRequest.getBucketName}.s3.amazonaws.com/${putObjectRequest.getKey}"
     }
 
-    S3Upload(new URI(uri))
+    S3UploadSuccess(new URI(uri), putObjectRequest.getKey)
```

Comment:
Might be better to use a `PutObjectResult` rather than `PutObjectRequest` as `PutObjectResult` implies the `PutObjectRequest` had succeeded.